app/src/main/java/org/mozilla/fenix/components/Core.kt

```diff
@@ -120,6 +120,7 @@ class Core(
             defaultSettings,
             GeckoProvider.getOrCreateRuntime(
                 context,
+                lazyAutofillStorage,
```

Comment:
This is not the right way to adding a storage delegate. Unfortunately, we have logins that started doing this with the pattern following.

Maybe we need to file an issue to fix this up to change these APIs before it happens the third or fourth time.

Reply:
This isn't adding the storage delegate. We're adding the storage delegate here https://github.com/mozilla-mobile/fenix/blob/8adc69f3fcfb73f83e02c87d62f038cf82ac4aa6/app/src/main/java/org/mozilla/fenix/gecko/GeckoProvider.kt#L93. This is simply about passing the underlying storage so that it can be passed to their respective storage delegate when the GeckoRuntime is created and can be set via `autocompleteStorageDelegate`, which is its proper GV API.

Reply:
We're talking about the same thing at a different abstraction level. See the Engine.kt API which is how we should be adding these storage delegates.

I think it's a bit too late in this particular process to clean this up, so let's do it in a follow up.